jwt-authorizer/templates/README.md

```diff
@@ -0,0 +1,162 @@
+# Deploying
+
+## Prerequisites
+This process assumes your ingress for the hostname has already been set up.
+We do not rely on tls secrets for any deployments under that assumptions
+
+### Getting Client Information
+
+If you do not have a client as secret, you first need to obtain one from CILogon OAuth2 Client ID and secret.
+
+Go here:
+https://cilogon.org/oauth2/register
+
+1. Add Client Name, e.g. "LSST LSP instance SSO"
+2. Contact Email
+3. Add hostname for Home URL
+  - `http://lsst-lsp-instance.example.com`
+4. Add callback URL for oauth2_proxy
+  - `http://lsst-lsp-instance.example.com/oauth2/callback`
+5. This is a private client
+
+6. Select Scopes:
+
+* email
+* profile
+* org.cilogon.userinfo
+
+7. Refresh Token Lifetime - 24 hours
+  - This is not really necessary, we can probably get by without refresh token
+
+Save that information.
+This is your client id and client secret.
+
+### After submission
+
+A separate email is required to CILogonhelp address to apply the client configuration
+from the client `cilogon:/client_id/6ca7b54ac075b65bccb9c885f9ba4a75` to your new
```

Comment:
What is this thing with the hash?

Reply:
This is specific to NCSA deployments - CILogon clients are generic. This is just a client id with a correct configuration which the NCSA/CILogon team can pull a configuration from. It was the only thing I was given as a way of conveying how that configuration should be applied to new clients.